lib/src/main/java/graphql/nadel/engine/transform/partition/NadelPartitionListMerger.kt

```diff
@@ -0,0 +1,45 @@
+package graphql.nadel.engine.transform.partition
+
+import graphql.nadel.engine.transform.result.NadelResultInstruction
+import graphql.nadel.engine.transform.result.json.JsonNode
+import graphql.nadel.engine.util.unwrapNonNull
+import graphql.normalized.ExecutableNormalizedField
+import graphql.schema.GraphQLList
+import graphql.schema.GraphQLObjectType
+import graphql.schema.GraphQLOutputType
+import graphql.schema.GraphQLScalarType
+
+object NadelPartitionListMerger {
+    fun mergeDataFromList(
+        dataFromPartitionCalls: List<Any?>,
+        thisNodesData: Any?,
+        parentNodes: List<JsonNode>,
```

Comment:
Maybe invoke `.single()` at the call site and pass in `parentNode: JsonNode` (singular)